tests/blackbox/testlib/utils.py

```diff
@@ -123,3 +125,67 @@ def exec_test_command(cmd):
         bytes(result[0]).decode("utf-8"),
         bytes(result[1]).decode("utf-8"),
     )
+
+
+class KernelKey:
+    """
+    A handle for operating on keys in the kernel keyring. The specified key will
+    be available for the lifetime of the test when used with the Python with
+    keyword and will be cleaned up at the end of the scope of the with block.
+    """
+
+    def __init__(self, key_data):
+        """
+        Initialize a key with the provided key data (passphrase).
+        :param bytes key_data: The desired key contents
+        :raises RuntimeError: if the keyctl command is not found in $PATH
+                              or a keyctl command returns a non-zero exit code
+        """
+        if shutil.which("keyctl") is None:
+            raise RuntimeError("Executable keyctl was not found in $PATH")
+
+        self.key_data = key_data
+
+    @staticmethod
+    def _raise_keyctl_error(return_code, args):
+        """
+        Raise an error if keyctl failed to complete an operation
+        successfully.
+        :param int return_code: Return code of the keyctl command
+        :param args: The command line that caused the command to fail
+        :type args: list of str
+        :raises RuntimeError
+        """
+        if return_code != 0:
+            raise RuntimeError(
+                "Command '%s' failed with exit code %s" % (" ".join(args), return_code)
+            )
+
+    def __enter__(self):
+        with open("/dev/urandom", "rb") as urandom_f:
+            key_desc = base64.b64encode(urandom_f.read(16)).decode("utf-8")
+
+        args = ["keyctl", "get_persistent", "@s", "0"]
+        exit_values = run(args, capture_output=True, text=True)
+        persistent_id = exit_values.stdout.strip()
```

Comment:
Please swap these two lines; dosn't make sense to get ```persistent_id``` if call has failed.

Reply:
Actually, do you even need ```persistent_id```? man page indicates that you could use ```@s```.

Reply:
I'll do a test to see if this is necessary, but this comes down to how we'd recommend users administer the keychain. I'll present both options once I have more information, but the way this is currently operating, it does work so the only benefit we will be getting out of this is a slightly simpler set of commands to recommend (which may still be worth it).

Reply:
It appears we do need `persistent_id`.

Reply:
Resolved.